test/FourBitTwoDisclosureDeviceUnlockerTest.java

```diff
@@ -1,31 +1,52 @@
 import org.junit.Assert;
 import org.junit.Test;
 
+import java.lang.reflect.Constructor;
 import java.lang.reflect.Field;
 import java.lang.reflect.InvocationTargetException;
 import java.lang.reflect.Method;
+import java.util.List;
 
+import static org.junit.Assert.fail;
 import static org.mockito.Mockito.*;
 
+@SuppressWarnings("ALL")
 public class FourBitTwoDisclosureDeviceUnlockerTest {
     @Test
     public void nullDeviceTest() {
         FourBitTwoDisclosureDeviceUnlocker.unlock(null);
     }
 
     @Test
+    public void getPermutationsTest() {
```

Comment:
I believe if you add throws Exception, or the more specific exceptions the method will throw and fail if you don't specify that you are expecting an exception. For the future it might be easier to just add the throw's instead of a try/catch. Either way is fine, just some advice :)